docs/api-spec.md

```diff
@@ -48,269 +72,447 @@ There is no formal specification of the Kubernetes API and Resource Model. This
+| `creationTimestamp` | string             | RECOMMENDED | If supported, `creationTimestamp` MUST be populated by the implementation, in [RFC3339](https://tools.ietf.org/html/rfc3339).                                                            |
+| `uid`               | string             | RECOMMENDED |                                                                                                                                                                                          |
+| `resourceVersion`   | string             | RECOMMENDED |                                                                                                                                                                                          |
+| `generation`        | int64              | RECOMMENDED |                                                                                                                                                                                          |
+| `namespace`         | string             | RECOMMENDED |                                                                                                                                                                                          |
+| `generateName`      | string             | RECOMMENDED | If supported by the implementation, when `generateName` is specified at creation, it MUST be prepended to a random string and set as the `name`, and not set on the subsequent response. |
 
-#### Metadata
+### TaskSpec
 
-Standard Kubernetes [meta.v1/ObjectMeta](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#objectmeta-v1-meta) resource.
+Defines the desired state of Task
 
-| Field Name           | Field Type         | Requirement |
-|----------------------|--------------------|-------------|
-| `name`               | string             | REQUIRED    |
-| `labels`             | map<string,string> | REQUIRED    |
-| `annotations`        | map<string,string> | REQUIRED    |
-| `creationTimestamp`* | string             | REQUIRED    |
-| `uid`                | string             | RECOMMENDED |
-| `resourceVersion`    | string             | RECOMMENDED |
-| `generation`         | int64              | RECOMMENDED |
-| `namespace`          | string             | RECOMMENDED |
-| `generateName`**     | string             | RECOMMENDED |
+| Field         | Type                                              | Requirement | Notes |
+| ------------- | ------------------------------------------------- | ----------- | ---- |
+| `description` | string                                            | RECOMMENDED |      |
+| `params`      | [][`ParamSpec`](#paramspec)                       | REQUIRED    |      |
+| `steps`       | [][`Step`](#step)                                 | REQUIRED    |      |
+| `sidecars`    | [][`Sidecar`](#sidecar)                           | RECOMMENDED |      |
```

Comment:
The lack of `sidecar` would break build `Tasks` that run docker in the `sidecar`. 
Any special reason for excluding it from the REQUIRED list?

Reply:
No, I wasn't sure if this feature was essential for most workloads though you are right that it is needed for docker builds. I moved it back to REQUIRED